### PR TITLE
feat: initial 'Remove account' implementation

### PR DESCRIPTION
**Pull request by david-allison**
## Purpose / Description
Google require that we more prominently display 'Delete Account'

## Fixes
* Some of #16256

----

* [x]  the path for requesting deletion is not prominently displayed (for example a little process diagram of "1) access this page 2) log in to the account 3) click delete account link" 4) confirm ?
    * [x] note that the destination URL of "remove-account" is not remembered for redirect after login if not authenticated which would be nice to have. So currently steps are "click remove-account link", "log in", "click account", "click remove account", "confirm", account is deleted...
* [ ] ⚠️ the name doesn't match - we are "AnkiDroid Flashcards" on Play Store listing but that is not listed on the page at all. This is difficult. We need to have that name on the Play Store, but the account is an "AnkiWeb" account based on the webpage. Is there a way to include some information on the deletion page that this account is used by all compatible Anki ecosystem apps including Anki Mobile, Anki Desktop, and AnkiDroid Flashcards?
* [x] Privacy Policy link

## Approach
* A 'remove account' button opens AnkiWeb
* A user must log in to AnkiWeb if they did not already do so in 'Shared Decks'
  * We perform an internal redirect back to the 'remove account' functionality
* A user must enter their password
* A user must type 'remove'

## How Has This Been Tested?
API 33 emulator

* with non-validated account
* with validated account

Screenshot 2024-04-23 at 19 36 40

Screenshot 2024-04-23 at 19 37 08

Screenshot 2024-04-23 at 19 57 21

Screenshot 2024-04-23 at 20 33 05

Screenshot 2024-04-23 at 20 33 13

## Learning (optional, can help others)
Google 🤷‍♂️ ... happily let copycats into the store, still make us jump through hoops

## Checklist
- [x] You have a descriptive commit message with a short title (first line, max 50 chars).
- [x] You have commented your code, particularly in hard-to-understand areas
- [x] You have performed a self-review of your own code
- [ ] UI changes: include screenshots of all affected screens (in particular showing any new or changed strings)
- [ ] UI Changes: You have tested your change using the [Google Accessibility Scanner](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor)

### CODE REVIEW

**Comment by mikehardy** [comment]
As long as the copycat also respects user privacy *and* we are not guarding intellectual property in the "Anki" name copycats are fine 🤷  - orthogonal issues

File: AnkiDroid/src/main/java/com/ichi2/anki/MyAccount.kt [hunk 1]
```diff
@@ -59,18 +59,16 @@ open class MyAccount : AnkiActivity() {
             STATE_LOGGED_IN -> {
                 val username = baseContext.sharedPrefs().getString("username", "")
                 usernameLoggedIn.text = username
-                toolbar = loggedIntoMyAccountView.findViewById(R.id.toolbar)
-                if (toolbar != null) {
-                    toolbar!!.title =
+                toolbar = loggedIntoMyAccountView.findViewById<Toolbar?>(R.id.toolbar)?.also { toolbar ->
```
Comment (mikehardy) [inline AnkiDroid/src/main/java/com/ichi2/anki/MyAccount.kt:74]: nitpick on commit message, you're using `?.also` not `?.let` in code but commit says let

File: AnkiDroid/src/main/res/values/01-core.xml [hunk 2]
```diff
@@ -293,4 +293,5 @@
     <string name="image_occlusion">Image Occlusion</string>
 
     <string name="remove_account" comment="open a website to start the deletion of an AnkiWeb account">Remove account</string>
+    <string name="privacy_policy">Privacy policy</string>
```
Comment (mikehardy) [inline AnkiDroid/src/main/res/values/01-core.xml:296]: out of curiosity - perhaps the Remove account and Privacy policy strings are already in anki upstream somewhere? Just curious - I haven't checked, but I would assume (dangerous) AnkiMobile needs this as well as those things are also required by Apple guidelines

Comment (david-allison) [reply AnkiDroid/src/main/res/values/01-core.xml]: That was an accidental push after a rebase.

We already have 'Privacy', don't have access to AnkiMobile strings AFAIK

File: AnkiDroid/src/main/java/com/ichi2/anki/MyAccount.kt [hunk 3]
```diff
@@ -130,6 +130,16 @@ open class MyAccount : AnkiActivity() {
         }
     }
 
+    /**
+     * Opens the AnkiWeb 'remove account' WebView
+     * @see R.string.remove_account_url
+     */
+    private fun openRemoveAccountScreen() {
+        // BUG: custom sync server doesn't use this URL
```
[the remark this answers is not in the document]
Comment (david-allison) [reply AnkiDroid/src/main/java/com/ichi2/anki/MyAccount.kt]: I don't want to take risks here, call me old-fashioned 🧐

**Comment by david-allison** [comment]
Nits picked, Privacy policy added

File: AnkiDroid/src/main/res/values/01-core.xml (hunk 2 above)
Comment (mikehardy) [reply AnkiDroid/src/main/res/values/01-core.xml]: I was thinking - maybe incorrectly - that anything in AnkiMobile would be in the translation blob we get from anki upstream. I say this without any inspection my part or knowledge to this point tho

**Review by mikehardy — APPROVED** [review]
Everything looks good, pulled it locally: tested all UI elements etc, working ✅ 
Fantastic, thank you

File: AnkiDroid/src/main/res/values/01-core.xml (hunk 2 above)
Comment (david-allison) [reply AnkiDroid/src/main/res/values/01-core.xml]: I believe we only ingest the desktop app currently

**Review by lukstbit — APPROVED** [review]
Ok, for the code changes.

>  Is there a way to include some information on the deletion page that this account is used by all compatible Anki ecosystem apps including Anki Mobile, Anki Desktop, and AnkiDroid Flashcards?

Ask Damien very nicely to put a mention about the clients currently using the account data(_Anki, Anki Mobile and AnkiDroid Flashcards_)?

**Comment by mikehardy** [comment]
Just trying to push strings through past flakiness... #16267

**Comment by dae** [comment]
> Ask Damien very nicely to put a mention about the clients currently using the account data(Anki, Anki Mobile and AnkiDroid Flashcards)?

I can do this as a last resort, but I think it's unreasonable for a third-party service to have to do this - have you tried pointing out to them that the service is not under your control? I'm sure Twitter/X's account removal page doesn't list all the third-party clients that access it for example.

**Comment by snowtimeglass** [comment]
As a desirable approach on the client (AnkiDroid) side, I suppose that it would be better to use AnkiWeb logo for the "AnkiWeb account" screen rather than AnkiDroid logo (only), to be less misleading. 
image


In case that an app has a screen for connecting to Twitter/X, it is natural to use Twitter/X logo in the screen, rather than the logo of the app itself (only). 



I think the same would be true for the screen for connecting to AnkiWeb.

**Comment by snowtimeglass** [comment]
> it would be better to use AnkiWeb logo for the "AnkiWeb account" screen rather than AnkiDroid logo (only), to be less misleading.

- Simply replacing the logo 
or 
- Displaying the two logos to visualize the independence of the app (AnkiDroid) and the external service (AnkiWeb) from each other.

  [Reference]
  

  [image]
  

  (Anyway, the current logo on the screen may be a bit too big.)

**Comment by david-allison** [comment]
> As a desirable approach on the client (AnkiDroid) side, I suppose that it would be better to use AnkiWeb logo for the "AnkiWeb account" screen rather than AnkiDroid logo (only), to be less misleading. 
> image
> 
> 
> In case that an app has a screen for connecting to Twitter/X, it is natural to use Twitter/X logo in the screen, rather than the logo of the app itself (only). 
> 
> 
> 
> I think the same would be true for the screen for connecting to AnkiWeb.
> 

Could you move these comments to a separate issue? I agree and they should be discussed, but they'll be lost if added to a closed PR

**Pull request merged**